core/domain/suggestion_services.py

```diff
@@ -0,0 +1,382 @@
+
+    Args:
+        author_id: str. The ID of the author of the suggestions.
+
+    Returns:
+        list(Suggestion). A list of suggestions by the given author.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_author(
+            author_id)]
+
+
+def get_suggestions_reviewed_by(reviewer_id):
+    """Gets a list of suggestions that have been reviewed by the given user.
+
+    Args:
+        reviewer_id: str. The ID of the reviewer of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions reviewed by the given user.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_reviewed_by(
+            reviewer_id)]
+
+
+def get_suggestions_assigned_to_reviewer(assigned_reviewer_id):
+    """Gets a list of suggestions assigned to the given user for review.
+
+    Args:
+        assigned_reviewer_id: str. The ID of the reviewer assigned to review the
+                suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions assigned to the given user
+            for review.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel
+        .get_suggestions_assigned_to_reviewer(assigned_reviewer_id)]
+
+
+def get_suggestions_by_status(status):
+    """Gets a list of suggestions with the given status.
+
+    Args:
+        status: str. The status of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions with the given status.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_status(
+            status)]
+
+
+def get_suggestion_by_type(suggestion_type):
+    """Gets a list of suggestions with the given type.
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+
+    Returns:
+        list(Suggestion). A list of suggestions of the given type.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_type(
+            suggestion_type)]
+
+
+def get_suggestions_by_target_id(target_type, target_id):
+    """Gets a list of suggestions to the entity with the given ID.
+
+    Args:
+        target_type: str. The type of target entity the suggestion is linked to.
+        target_id: str. The ID of the target entity the suggestion is linked to.
+
+    Returns:
+        list(Suggestion). A list of suggestions linked to the entity.
+    """
+    return [
+        get_suggestion_from_model(s)
+        for s in suggestion_models.SuggestionModel.get_suggestions_by_target_id(
+            target_type, target_id)]
+
+
+def _update_suggestion(suggestion):
+    """Updates the given suggestion.
+
+    Args:
+        suggestion: Suggestion. The domain object of the suggestion to be
```

Comment:
The suggestion to be updated.

(When talking about nouns the default assumption is that they're domain objects.)

Reply:
Done